ni_measurementlink_service/_internal/parameter/serialization_strategy.py

```diff
@@ -229,6 +295,12 @@ def get_decoder(
             Callable[[int], Callable]: ScalarDecoder or VectorDecoder.
 
         """
+        if message_type == xydata_pb2.DoubleXYData.DESCRIPTOR.full_name:
```

Comment:
I think this is called for each request, so we should move the message type comparisons (of which there will be more in the future) out of the hot path.

Example:
```
decoder_mapping = Context._FIELD_TYPE_TO_DECODER_MAPPING.get(type)
if decoder_mapping is not None:
    scalar_decoder, array_decoder = decoder_mapping
    return array_decoder if repeated else scalar_decoder
elif message_type == blah blah blah:
elif message_type == blah blah blah2:
elif message_type == blah blah blah3:
else:
    raise Exception(f"Error can not decode type '{type}'")
```

Reply:
Also, a 2nd dict of message_type -> decoder would make sense.

Reply:
Deferring the addition of a message_type -> decoder dictionary until more messages are added.